src/app/[locale]/_hooks/useToken.tsx

```diff
@@ -1,31 +1,37 @@
 'use client';
 import { useEffect, useState } from 'react';
-import { extractToken, parseJwt, userFromJwtToken } from '../_utils/jwt';
-import { cookieTokenOps, cookieUserOps } from '../_utils/cookie';
+import { storageTokenOps, storageUserOps } from '../_utils/storage';
 
-interface Token {
+interface IToken {
   token: string;
-  tokenError: 'PENDING' | 'OK' | 'ERROR';
+  isTokenValid: () => boolean | undefined;
+  removeToken: () => void;
 }
 
-const useToken = (): Token => {
+const useToken = (): IToken => {
   const [token, setToken] = useState<string>('');
-  const [tokenError, setTokenError] = useState<'PENDING' | 'OK' | 'ERROR'>('PENDING');
 
   useEffect(() => {
-    if (parseJwt(extractToken())) {
-      setToken(extractToken());
-      setTokenError('OK');
-      cookieTokenOps.write(extractToken());
-      cookieUserOps.write(userFromJwtToken(extractToken()));
-    } else {
-      setTokenError('ERROR');
+    if (typeof window !== 'undefined') {
```

Comment:
You are using `typeof windows !== 'undefined'` many times. What do you think about writing a more epxlicit utility function?

Reply:
created isBrowser common utility function